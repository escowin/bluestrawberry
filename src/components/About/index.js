import React from "react";

function About() {
    const copyrightYear = new Date().getFullYear()

    return (
        <footer className="flex-column">
                <h2 className="header">About bluestrawberry</h2>
                <p>wordless iteration</p>
                <p id="copyright">
                    &copy; {copyrightYear}
                    <a href="/"> edwin m. escobar</a>
                </p>
        </footer>
    )
}

export default About;