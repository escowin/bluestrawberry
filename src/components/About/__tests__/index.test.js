import React from "react";
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import About from "..";

describe('About component', () => {
    // render About test
    // - first test
    it('renders', () => {
        render(<About/>);
    })
    // - second test
    it('matches snapshot DOM node structure', () => {
        // asFragment returns component snapshot
        const { asFragment } = render(<About/>);
        // tests whether expected and actual outcomes match
        // enter u in cli to update snapshot if component has been updated since last test
        expect(asFragment()).toMatchSnapshot();
    })
})
afterEach(cleanup);