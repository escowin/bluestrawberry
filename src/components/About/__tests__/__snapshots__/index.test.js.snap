// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <footer
    class="flex-column"
    id="about"
  >
    <h2>
      About bluestrawberry
    </h2>
    <p>
      iterations of the wordless narrative
    </p>
    <p
      id="copyright"
    >
      Â© 2022
      <a
        href="/"
      >
         edwin m. escobar
      </a>
    </p>
  </footer>
</DocumentFragment>
`;
